### PR TITLE
Separate workspace validators

### LINKED ISSUE #11035: Separate workspace validators into individual implementation files
This issue was originally [TRAC 10193](http://trac.mantidproject.org/mantid/ticket/10193)

... and add separate unit tests for them in API. 

The WorkspaceValidators are all currently implemented in the same header
file, ```API/inc/MantidAPI/WorkspaceValidators.h```, with the tests all in a single unit test file in DataObjects!

We should separate them out like all other classes into single implementation headers and source files + tests in the API layer using the ```WorkspaceTester``` class in ```MantidTestHelpers/FakeObjects.h``` instead of a "real" workspace type. 

 - - - - 
Keywords: Maintenance

### PR DESCRIPTION
**Pull request by mantid-roman**
Fixes  #11035.

To test review the changes.

### CODE REVIEW

File: Framework/API/test/CommonBinsValidatorTest.h [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  }
+  static void destroySuite(CommonBinsValidatorTest *suite) { delete suite; }
+
+  void test_empty() {
+    auto ws = boost::make_shared<WorkspaceTester>();
+    CommonBinsValidator validator;
+    TS_ASSERT_EQUALS(validator.isValid(ws), "");
+  }
+
+  void xtest_zero_length_bins() {
```
Comment (FedeMPouzols) [inline Framework/API/test/CommonBinsValidatorTest.h]: Is this test disabled (`xtest`) for any particular reason. I tried to run it and apparently it passes as it should.

**Comment by FedeMPouzols** [comment]
This looks ready to be merged, all the validators have been moved into their individual headers, the new/split tests added, and all the headers updated all throughout algorithms.
I only saw the disabled test that I commented inline.

File: Framework/API/test/CommonBinsValidatorTest.h (hunk 1 above)
Comment (mantid-roman) [inline Framework/API/test/CommonBinsValidatorTest.h]: I forgot to enable it

**Comment by FedeMPouzols** [comment]
Not bothering to wait for the system tests. This was already green and all unit/doc tests have passed once more.

**Pull request merged**